src/trigger/progress/threshold/mod.rs

```diff
@@ -0,0 +1,99 @@
+pub mod amount_type;
+mod threshold_type;
+
+use crate::{
+    render_util::{enum_combo, helper, input_u32},
+    traits::RenderOptions,
+};
+use amount_type::AmountType;
+use nexus::imgui::{ComboBoxFlags, Ui};
+use serde::{Deserialize, Serialize};
+use threshold_type::ThresholdType;
+#[derive(Debug, Default, Clone, Serialize, Deserialize)]
+pub struct ProgressThreshold {
+    /// Threshold type.
+    pub threshold_type: ThresholdType,
+
+    /// Amount type.
+    #[serde(default)]
+    pub amount_type: AmountType,
+}
+
+impl ProgressThreshold {
+    pub fn is_met(&self, progress: u32) -> bool {
+        match self.threshold_type {
+            ThresholdType::Always => true,
+            ThresholdType::Present => progress > 0,
+            ThresholdType::Missing => progress == 0,
+            ThresholdType::Min(required) => progress >= required,
+            ThresholdType::Max(required) => progress <= required,
+            ThresholdType::Exact(required) => progress == required,
+            ThresholdType::Between(min, max) => (min..=max).contains(&progress),
+        }
+    }
+}
+
+impl RenderOptions for ProgressThreshold {
+    fn render_options(&mut self, ui: &Ui) {
+        ui.group(|| {
+            enum_combo(
+                ui,
+                "Threshold",
+                &mut self.threshold_type,
+                ComboBoxFlags::empty(),
+            );
+            helper(ui, || ui.text("When to display"));
+
+            let (input_label, step, step_fast);
+            match self.amount_type {
+                AmountType::Intensity => {
+                    input_label = "Intensity";
+                    step = 1;
+                    step_fast = 10;
+                }
+                AmountType::Duration => {
+                    input_label = "Duration (ms)";
```

Comment:
Seconds might be the more intuitive unit here. Explanations about the unit could be in a helper text.